util.go

```diff
@@ -50,19 +53,180 @@ func parseTag(path string) string {
 	return strings.TrimSuffix(filepath.Base(path), ".toml")
 }
 
-func parseDependencies(r io.Reader) ([]dependency, error) {
-	var deps []dependency
+func parseDependencies(commit string) ([]dependency, error) {
+	rd, err := fileFromRev(commit, vendorConf)
+	if err == nil {
+		return parseVendorConfDependencies(rd)
+	}
+	rd, err2 := fileFromRev(commit, goMod)
+	if err2 == nil {
+		return parseGoModDependencies(rd)
+	}
+	return nil, errors.Errorf("finding current dep file failed. vendor.conf error: %v, go.mod error: %v", err, err2)
+}
+
+func parseGoModDependencies(r io.Reader) ([]dependency, error) {
+	var err error
+
+	depMap := make(map[string]*dependency)
+	replaceMap := make(map[string]string)
 	s := bufio.NewScanner(r)
 	for s.Scan() {
-		ln := strings.TrimSpace(s.Text())
-		if strings.HasPrefix(ln, "#") || ln == "" {
+		ln := sanitizeLine(s.Text(), "//")
+		if ln == "" {
 			continue
 		}
-		cidx := strings.Index(ln, "#")
-		if cidx > 0 {
-			ln = ln[:cidx]
+		parts := strings.Fields(ln)
+
+		// scan the file until we find `$DIRECTIVE (`
+		if len(parts) == 2 && parts[1] == "(" {
+			if parts[0] == "require" {
+				depMap, err = processRequiresSection(s)
+				if err != nil {
+					return nil, err
+				}
+			} else if parts[0] == "replace" {
+				replaceMap, err = processReplaceSection(s)
```

Comment:
It looks like the `replace` directive is not always a section

See https://github.com/moby/buildkit/blob/master/go.mod#L77

That may just be from go 1.11, not sure if the syntax has solidified since then.

Reply:
I added support for directives outside of a section!